### ISSUE TITLE
[6.0.1] initialValues for react-widgets elements

### ISSUE THREAD

**Issue by KseniyaT** [erikras/redux-form#1645]
Hello!
I use redux-form v.6.0.1 with react-widgets.

Now I have problem with initializing from state.

```
import React , {Component, PropTypes} from 'react';
import {render}                       from 'react-dom';
import {Field, reduxForm}             from 'redux-form';
import {connect}                      from 'react-redux';
import InputField                     from '{path_to_InputField}';
import * as actions                   from '{path_to_actions}';
...

const validate = values => {
  const errors = {};
  if (!values.username) {
    errors.username = 'Required field'
  } else if (values.username && values.username.length > 15) {
    errors.username = 'Must be less than 15 characters'
  }
  ...
  return errors;
};

class PersonalDataForm extends Component {
  constructor(props) {
    super(props);
  }
  render(){
    const { initialValues, handleSubmit, submitting } = this.props;

    return (
      <form
        className='form'
        onSubmit={handleSubmit(actions.savePersonalData)}
      >
            <InputField
              name='username'
              type='text'
              placeholder='Username'
            />
            <InputField
              name='firstName'
              type='text'
              placeholder='First name'
            />
            <InputField
              name='lastName'
              type='text'
              placeholder='Last name'
            />
            ...
          <button type='submit' disabled={submitting} >
            {'Save'}
          </button>
           ...
      </form>
    )
  }
}

PersonalDataForm = reduxForm({
  form: 'personalData',
  validate
})(PersonalDataForm);

PersonalDataForm = connect(
  state => ({
    initialValues: {
      username: 'SomeUserName',
      lastName: state.personal_data.personalData.lastName,
      firstName: state.personal_data.personalData.firstName
      ...
    }
  })
)(PersonalDataForm);

export default PersonalDataForm;
```

So, now only username is existing initial value, other values are undefined.

InputField code:

```
import React, {Component, PropTypes}  from 'react';
import {render}                       from 'react-dom';
import {Field}                        from 'redux-form';

const renderField = ({ input, label, type, placeholder, normalize, className, meta: { touched, error } }) => (
  <div className='form__group'>
    <label className='form__label'>{label}</label>
    <div className='relative'>
      <input
        {...input}
        placeholder={placeholder}
        type={type}
        normalize={normalize}
        className={className? (className +' form__control'): 'form__control'}
      />
    </div>
    {touched && error && <span className="error-mark">{error}</span>}
  </div>
);

class InputField extends Component {
  constructor(props) {
    super(props);
  }
  render(){
    return (
      <Field
        component={renderField}
        name={this.props.name}
        type={this.props.type}
        placeholder={this.props.placeholder}
        label={this.props.placeholder}
        className={this.props.className}
        normalize={this.props.normalize}
      />
    )
  }
}

export default InputField;
```

Value firstName in a console after load is:
_this.props.initialValues.firstName: undefined
...
this.props.initialValues.firstName: NameTest_

**Comment by Sir4ur0n** [comment]
Are you using v6.0.1 of redux-form?
Can you please post the code of `InputField`? It might be that you don't correctly pass the `name` prop to a redux-form `Field` component, but instead hard coded it in `InputField`. It's just a guess without the code though.

**Comment by KseniyaT** [comment]
@Sir4ur0n thank you for your answer, I updated the question. I think that "name" props was passed correctly, and everything is working when I pass String to initialValues, but it does not work with data from state. So, I think the cause is different here. :-)

**Comment by Sir4ur0n** [comment]
I think I understand what's happening. 
TLDR: Pass `enableReinitialize=true` to your form or prevent rendering until the state is fully loaded.

Long story: As explained in the [initialValues example](http://redux-form.com/6.0.1/examples/initializeFromState/), this only works the first time. As on initial load, you have an initial value which is undefined (`username: 'SomeUserName'`), this counts as form initialization. Later when the fields `lastName` and `firstName` are defined, it's too late, the form was already initialized. 
The easy solution is to pass `enableReinitialize=true` to your form, as explained in the [reduxForm documentation](http://redux-form.com/6.0.1/docs/api/ReduxForm.md/). It means the form will initialize with the username, and then re-initialize once the `lastName` and `firstName` props become defined.
Another solution is to prevent rendering until the state is fully loaded:
```javascript
render() {
  if (this.props.firstName && this.props.lastName) {
    // Your current render code here
  } else {
    return null; // <- This prevents rendering
  }
}
```

**Comment by KseniyaT** [comment]
@Sir4ur0n many thanks!It works for me! (enableReinitialize=true)

**Comment by gustavohenke** [comment]
Because this issue saw almost no activity for more than a year, I'm closing it.
